assets/app/views/edit/build-config.html

```diff
@@ -0,0 +1,519 @@
+                                ng-change="updateOutputImageStreams(options.pickedPushToNamespace, true)"
+                                required>
+                              </select>
+                              <div class="warnings">
+                                <span class="text-danger" ng-if="!availableProjects.contains(options.pickedPushToNamespace)">Selected Namespace is not available.</span>
+                              </div>
+                            </div>
+                          </div>
+                          <div class="col-lg-4">
+                            <div class="label-with-selector">
+                              <label class="multiselector-label">Image Stream</label>
+                              <select
+                                class="form-control multiselector select-output-name"
+                                ng-model="options.pickedPushToImageStream"
+                                ng-options="name as name for name in pushTo.imageStreams track by name"
+                                ng-change="clearSelectedOutputTag()"
+                                required>
+                              </select>
+                              <div class="warnings">
+                                <span class="text-danger" ng-if="pushTo.imageStreams.length === 0">Selected Namespace doesn't contain any Image Streams to build from.</span>
+                                <span class="text-danger" ng-if="pushTo.imageStreams.length !== 0 && !pushTo.imageStreams.contains(options.pickedPushToImageStream)">Selected Namespace doesn't contain Image Stream <b>'{{options.pickedPushToImageStream}}'</b>.</span>
+                              </div>
+                            </div>
+                          </div>
+
+                          <div class="col-lg-4">
+                            <div class="label-with-selector">
+                              <label class="multiselector-label">Tag</label>
+                              <div>
+                              <input class="form-control multiselector select-output-tag"
+                                id="FromTypeLink"
+                                name="FromTypeLink"
+                                type="text"
+                                ng-model="options.pickedPushToImageStreamTag"
+                                ng-disabled="options.pickedPushToImageStream === ''"
+                                autocorrect="off"
+                                autocapitalize="off"
+                                spellcheck="false"
+                                required>
+                              </div>
+                              <div class="warnings">
+                                <span class="text-warning" ng-if="pushTo.tags[options.pickedPushToImageStream].contains(options.pickedPushToImageStreamTag)">Tag <b>'{{options.pickedPushToImageStreamTag}}'</b> is already created and will be overridden on build.</span>
```

Comment:
I'd hide this warning if the namespace, image stream, and tag haven't been changed. If you give each field a `name` attribute, you can test `form.myField.$dirty` for each.

You might pull the logic into a scope method if it gets too long.

```
scope.showOutputTagWarning = function() {
 ...
};
```

and

```
<span class="text-warning" ng-if="showOutputTagWarning()">...</span>
```